Source/Core/Common/FilesystemWatcher.cpp

```diff
@@ -13,51 +13,65 @@ namespace Common
 FilesystemWatcher::FilesystemWatcher() = default;
 FilesystemWatcher::~FilesystemWatcher() = default;
 
-void FilesystemWatcher::Watch(const std::string& path)
+void FilesystemWatcher::Watch(const std::string& path, StartedCallback started_callback)
 {
-  const auto [iter, inserted] = m_watched_paths.try_emplace(path, nullptr);
-  if (inserted)
+  auto& watch_ptr = m_watched_paths[path];
+  // Are we already watching this path?
+  if (watch_ptr != nullptr)
   {
-    iter->second = std::make_unique<wtr::watch>(path, [this](wtr::event e) {
-      const auto watched_path = PathToString(e.path_name);
-      if (e.path_type == wtr::event::path_type::watcher)
-      {
-        if (watched_path.starts_with('e'))
-          ERROR_LOG_FMT(COMMON, "Filesystem watcher: '{}'", watched_path);
-        else if (watched_path.starts_with('w'))
-          WARN_LOG_FMT(COMMON, "Filesystem watcher: '{}'", watched_path);
-        return;
-      }
+    // The caller doesn't require notification on start so we're done.
+    if (!started_callback)
+      return;
 
-      if (e.effect_type == wtr::event::effect_type::create)
-      {
-        const auto unified_path = WithUnifiedPathSeparators(watched_path);
-        PathAdded(unified_path);
-      }
-      else if (e.effect_type == wtr::event::effect_type::modify)
+    // Remove and recreate the watch object so we can wait for a new "started" event.
```

Comment:
This feels potentially dangerous.  If we are watching a path and we call watch again on the same path, the current watcher would be uninitialized and we could miss events.  Is there a scenario you had in mind where this was advantageous?

Reply:
But we re-do `Common::DoFileSearch` after `Watch` is called, yeah?